### PR TITLE
[API] Remove error field from data field in API responses

### PR DESCRIPTION
**Pull request by CarlosRS9**
Hello team,

This PR updates the API endpoints to avoid having the newly-added `error` field inside the `data` field, if present.

Here are the list of affected endpoints and an example of their response after this fix:

**POST /agents**
```
{
  "data": {
    "id": "001",
    "key": "MDAxIENhcmxvcyAxMC4wLjAuOSAxMzViNmRjZWJiOTQ5NzM1YjVlOGI1OGQwZGMwNTRlY2VmYWM3ODhhMWYyOWQ1YTUxMTVjMDdjMTZmN2IwY2E5"
  },
  "error": 0
}
```

**GET  /agents/{agent_id}/config/{component}/{configuration}**
```
{
  "data": {
    "global": {
      "email_notification": "no",
      "logall": "no",
      "logall_json": "no",
      "integrity_checking": 8,
      "rootkit_detection": 8,
      "host_information": 8,
      "prelude_output": "no",
      "zeromq_output": "no",
      "jsonout_output": "yes",
      "alerts_log": "yes",
      "stats": 4,
      "memory_size": 8192,
      "white_list": [
        "127.0.0.1",
        "212.230.135.1",
        "212.230.135.2",
        "localhost.localdomain"
      ],
      "rotate_interval": 0,
      "max_output_size": 0
    }
  },
  "error": 0
}
```

**POST /agents/insert/quick**
```
{
  "data": {
    "id": "003",
    "key": "MDAzIHN0cmluZyBhbnkgNGFhYzBjZjNiZWRmOGE4MzAxOWE3ZDU0OTk2OWM1N2M1YzVlZmY4Yzk2MzEwMmUxOWY1MDUyNDEwZTU4NDViMg=="
  },
  "error": 0
}
```

**GET /groups/{group_id}/configuration**
```
{
  "data": {
    "total_affected_items": 1,
    "affected_items": [
      {
        "filters": {},
        "config": {}
      }
    ]
  },
  "error": 0
}
```

**GET /groups/{group_id}/files/{file_name}/json**
```
[
  {
    "filters": {},
    "config": {}
  }
]
```

**POST /agents/insert**
```
{
  "data": {
    "id": "123",
    "key": "MTIzIE5ld0hvc3RfMiAxMC4wLjEwLjExIDFhYmNkZWZnaGlqa2xtbm9wcXJzdHV2d3h5emFiY2RlZmdoaWprbG1ub3BxcnN0dXZ3eHl6YWJjZGVmZ2hpNjQ="
  },
  "error": 0
}
```

**GET /agents/summary/status**
```
{
  "data": {
    "active": 1,
    "disconnected": 0,
    "never_connected": 4,
    "pending": 0,
    "total": 5
  },
  "error": 0
}
```

**GET /cluster/status**
```
{
  "data": {
    "enabled": "yes",
    "running": "yes"
  },
  "error": 0
}
```

**GET /**
```
{
  "data": {
    "title": "Wazuh API REST",
    "api_version": "4.0.0",
    "revision": 4000,
    "license_name": "GPL 2.0",
    "license_url": "https://github.com/wazuh/wazuh/blob/master/LICENSE",
    "hostname": "wazuh-master",
    "timestamp": "2020-10-07T08:32:11+0000"
  },
  "error": 0
}
```

**GET /overview/agents**
```
{
  "data": {
    "nodes": [
      {
        "count": 1,
        "node_name": "master-node"
      },
      {
        "count": 4,
        "node_name": "unknown"
      }
    ],
    "groups": [
      {
        "name": "default",
        "count": 0,
        "mergedSum": "fd756ba04d9c32c8848d4608bec41251",
        "configSum": "ab73af41699f13fdd81903b5f23d8d00"
      }
    ],
    "agent_os": [
      {
        "os": {
          "name": "Ubuntu",
          "platform": "ubuntu",
          "version": "18.04.5 LTS"
        },
        "count": 1
      },
      {
        "os": {
          "name": "unknown",
          "platform": "unknown",
          "version": "unknown"
        },
        "count": 4
      }
    ],
    "agent_status": {
      "data": {
        "active": 1,
        "disconnected": 0,
        "never_connected": 4,
        "pending": 0,
        "total": 5
      },
      "error": 0
    },
    "agent_version": [
      {
        "count": 1,
        "version": "Wazuh v4.0.0"
      },
      {
        "count": 4,
        "version": "unknown"
      }
    ],
    "last_registered_agent": [
      {
        "status": "never_connected",
        "dateAdd": "2020-10-07T08:21:22Z",
        "name": "NewHost_2",
        "registerIP": "10.0.10.11",
        "id": "123",
        "node_name": "unknown",
        "ip": "10.0.10.11"
      }
    ]
  },
  "error": 0
}
```

**GET /security/user/authenticate**
```
{
  "data": {
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjAyMDYxMDY4LCJleHAiOjE2MDIwNjQ2NjgsInN1YiI6IndhenVoIiwicmJhY19yb2xlcyI6WzFdLCJyYmFjX21vZGUiOiJ3aGl0ZSJ9.f_GmCHcPNCT2HP4vXJzLT5h9BXTcT7zauVn0gwBQaIU"
  },
  "error": 0
}
```

**GET /security/resources**
```
{
  "data": {
    "*:*": {
      "description": "Resource applied in functions which take part in the creation of a specific resource. We call these functions, resourceless functions. Example: agent:create"
    },
    "agent:id": {
      "description": "Reference agents via agent ID (i.e. agent:id:001)"
    },
    "group:id": {
      "description": "Reference agent groups via group ID (i.e. group:id:default)"
    },
    "node:id": {
      "description": "Reference cluster nodes via node ID (i.e. node:id:worker1)"
    },
    "file:path": {
      "description": "Reference files via its path (i.e. file:path:etc/rules/new_rule.xml)"
    },
    "decoder:file": {
      "description": "Reference decoder files via its path (i.e. decoder:file:0005-wazuh_decoders.xml)"
    },
    "list:path": {
      "description": "Reference list files via its path (i.e. list:path:etc/lists/audit-keys)"
    },
    "rule:file": {
      "description": "Reference rule files via its path (i.e. rule:file:0610-win-ms_logs_rules.xml)"
    },
    "policy:id": {
      "description": "Reference security policies via its id (i.e. policy:id:1)"
    },
    "role:id": {
      "description": "Reference security roles via its id (i.e. role:id:1)"
    },
    "user:id": {
      "description": "Reference security users via its id (i.e. user:id:1)"
    },
    "rule:id": {
      "description": "Reference security rules via its id (i.e. rule:id:1)"
    }
  },
  "error": 0
}
```

GET /security/actions**
```
{
  "data": {
    "active-response:command": {
      "description": "Allow to execute active response commands in the agents",
      "resources": [
        "agent:id"
      ],
      "example": {
        "actions": [
          "active-response:command"
        ],
        "resources": [
          "agent:id:001"
        ],
        "effect": "allow"
      },
      "related_endpoints": [
        "PUT /active-response"
      ]
    },
    
    · · ·

    "security:update_config": {
      "description": "Update current security configuration",
      "resources": [
        "*:*"
      ],
      "example": {
        "actions": [
          "security:update_config"
        ],
        "resources": [
          "*:*:*"
        ],
        "effect": "allow"
      },
      "related_endpoints": [
        "PUT /security/config",
        "DELETE /security/config"
      ]
    }
  },
  "error": 0
}
```

**GET /security/config**
```
{
  "data": {
    "auth_token_exp_timeout": 3600,
    "rbac_mode": "white"
  },
  "error": 0
}
```

### CODE REVIEW

File: framework/wazuh/cluster.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def get_status_json():
 
     :return: Dictionary with the cluster status.
     """
-    return get_cluster_status()
+    return {'data': get_cluster_status(), 'error': 0}
```
Comment (davidjiglesias) [inline framework/wazuh/cluster.py:60]: Why not use WazuhResult?

File: framework/wazuh/agent.py [hunk 2]
```diff
@@ -861,7 +862,7 @@ def get_full_overview() -> WazuhResult:
     except IndexError:  # an IndexError could happen if there are not registered agents
         last_registered_agent = []
     # combine results in an unique dictionary
-    result = {'nodes': stats_distinct_node, 'groups': groups, 'agent_os': stats_distinct_os, 'agent_status': summary,
-              'agent_version': stats_version, 'last_registered_agent': last_registered_agent}
+    result = {'data': {'nodes': stats_distinct_node, 'groups': groups, 'agent_os': stats_distinct_os, 'agent_status': summary,
+              'agent_version': stats_version, 'last_registered_agent': last_registered_agent}}
 
     return WazuhResult(result)
```
Comment (davidjiglesias) [inline framework/wazuh/agent.py:868]: Modify the WazuhResult return but don't add data to result variable

File: framework/wazuh/security.py [hunk 3]
```diff
@@ -955,13 +955,13 @@ def get_rbac_actions(endpoint: str = None):
...
 
-    return WazuhResult(data)
+    return WazuhResult({'data': data})
 
 
 @expose_resources(actions=['security:read_config'], resources=['*:*:*'])
 def get_security_config():
     """Returns current security configuration."""
-    return configuration.security_conf
+    return {'data': configuration.security_conf }
```
Comment (davidjiglesias) [inline framework/wazuh/security.py:964]: WazuhResult()?

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: framework/wazuh/agent.py [hunk 4]
```diff
@@ -65,16 +65,16 @@ def get_agents_summary_status(agent_list=None):
     :param agent_list: List of agents ID's.
     :return: WazuhResult.
     """
-    result = WazuhResult({'active': 0, 'disconnected': 0, 'never_connected': 0, 'pending': 0, 'total': 0})
+    result = WazuhResult({'data': {'active': 0, 'disconnected': 0, 'never_connected': 0, 'pending': 0, 'total': 0}})
```
[the remark this answers is not in the document]
Comment (CarlosRS9) [reply framework/wazuh/agent.py]: Done

File: framework/wazuh/cluster.py (hunk 1 above)
Comment (CarlosRS9) [reply framework/wazuh/cluster.py]: Changed

File: framework/wazuh/agent.py (hunk 2 above)
Comment (CarlosRS9) [reply framework/wazuh/agent.py]: done

File: framework/wazuh/security.py (hunk 3 above)
Comment (CarlosRS9) [reply framework/wazuh/security.py]: changed

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**